src/main/java/org/webjars/WebJarVersionLocator.java

```diff
@@ -47,47 +68,86 @@ public WebJarVersionLocator() {
      *
-     * @param webJarName The name of the WebJar, i.e. bootstrap
-     * @param exactPath The path to the file within the WebJar, i.e. js/bootstrap.js
-     * @return The path to the file in the standard WebJar classpath location, including the version, i.e. bootstrap/3.1.1/js/bootstrap.js
+     * <pre>{@code
+     * // returns "bootstrap/3.1.1/js/bootstrap.js"
+     * locator.path("bootstrap", "css/bootstrap.css");
+     *
+     * // returns "bootstrap/3.1.1/js/bootstrap.js" as well
+     * locator.path("bootstrap", "3.1.1/css/bootstrap.css");
+     *
+     * // returns null, assuming there is no "unknown" WebJar
+     * locator.path("unknown", "some/file.css");
+     * }</pre>
+     *
+     * <p><b>Note:</b> When the {@code filePath} starts with a version, that is not the correct version of the WebJar, this method returns a path with both versions.
+     * <br>For example {@code bootstrap/5.3.3/3.1.1/css/boostrap.css}.
+     *
+     * <p><b>Note:</b> This method does not perform any checks if the resulting path references an existing file.
+     *
+     * @param webJarName The name of the WebJar
+     * @param filePath   The path to the file within the WebJar
+     * @return The versioned path relative to the standard WebJar classpath location; null if no version for {@code webJarName} is known
+     * @see WebJarVersionLocator#fullPath(String, String)
+     * @see WebJarVersionLocator#WEBJARS_PATH_PREFIX
      */
     @Nullable
-    public String path(final String webJarName, final String exactPath) {
+    public String path(final String webJarName, final String filePath) {
         final String version = version(webJarName);
 
-        if (!isEmpty(version)) {
-            return String.format("%s/%s/%s", webJarName, version, exactPath);
+        if (notEmpty(version)) {
+            if (filePath.startsWith(version)) {
+                return String.format("%s/%s", webJarName, filePath);
+            } else {
+                return String.format("%s/%s/%s", webJarName, version, filePath);
+            }
         }
 
         return null;
     }
 
     /**
-     * @param webJarName The name of the WebJar, i.e. bootstrap
-     * @return The version of the WebJar, i.e 3.1.1
+     * This method tries to determine the available version for a WebJar in the classpath.
+     *
+     * <p>For official WebJars, the version lookup is performed by checking for a {@code pom.properties} file for either {@link WebJarVersionLocator#NPM}
+     * or {@link WebJarVersionLocator#PLAIN} WebJars within {@code META-INF/maven}. The lookup result is cached.
+     *
+     * <p>Custom WebJars can be registered by using a {@code webjars-locator.properties} file. See {@link WebJarVersionLocator} for details.
+     *
+     * @param webJarName The name of the WebJar (the Maven artifact ID)
```

Comment:
Thanks for pointing that out. How about:

```suggestion
     * @param webJarName The directory name in the WebJar following `META-INF/resources/webjars` - Usually the same as the Maven artifact ID
```

Reply:
Inspried by this suggestion I reworded the description a bit and added it to the other two methods.

Hope it is ok the way it is now. :)